pure.zsh

```diff
@@ -691,6 +694,12 @@ prompt_pure_state_setup() {
 	)
 }
 
+# Return true if executing inside a Docker or LXC container.
+prompt_pure_is_inside_container() {
+	( [[ -r /proc/1/cgroup ]] && (( $(grep --count --extended-regexp "(lxc|docker)" /proc/1/cgroup ) > 0 )) ) ||
+		(( $(env | grep --count 'container=lxc') > 0 ))
```

Comment:
```suggestion
	([[ -r /proc/1/cgroup ]] && grep -q -E "(lxc|docker)" /proc/1/cgroup) \
		|| env | grep -q 'container=lxc'
```

Proposing a few simplifications and compatibility improvements here:

- `--extended-regexp` is not supported on `busybox` (often used in Docker), `-E` is more compatible
- `-q` keeps grep silent and we can simply rely on it's exit status instead of checking count
- Speed up execution by not using subshells to capture output

Reply:
Another though, can we simply check `container=lxc` vi environment variable instead? Would save us from the execution of env/grep. (Not running LXC myself so can't test.)

Reply:
Thank you for the suggestions.  I'm having zplug troubles so I will try out a different plugin manager so I can test this efficiently before re-requesting review.

Reply:
Done.

Reply:
Noticed the grep is still running in a subshell `$( )`, you can remove that still.

Reply:
Good catch.  I removed it.